cmd/thanos/flags.go

```diff
@@ -36,6 +36,16 @@ func regCommonServerFlags(cmd *kingpin.CmdClause) (*string, *string, func(log.Lo
 
 	refreshInterval := cmd.Flag("cluster.refresh-interval", "Interval for membership to refresh cluster.peers state, 0 disables refresh.").Default(cluster.DefaultRefreshInterval.String()).Duration()
 
+	secretKey := cmd.Flag("cluster.secret-key", "Initial secret key to communicate cluster with encryption. Could be AES-128, AES-192, or AES-256 (hexadecimal data only)").HexBytes()
```

Comment:
Probably worth mentioning the gossip protocol, since it's mentioned in other flags and it makes it clear exactly what this key is used for. For example:

> Initial secret key to encrypt cluster gossip. Can be one of AES-128, AES-192, or AES-256 in hexadecimal format.